src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/ReplicationCheckerTest.java

```diff
@@ -0,0 +1,158 @@
+import uk.gov.hmcts.reform.wataskmanagementapi.db.TCExtendedContainerDatabaseDriver;
+import uk.gov.hmcts.reform.wataskmanagementapi.entity.TaskHistoryResource;
+import uk.gov.hmcts.reform.wataskmanagementapi.entity.TaskResource;
+import uk.gov.hmcts.reform.wataskmanagementapi.repository.TaskResourceRepository;
+import uk.gov.hmcts.reform.wataskmanagementapi.services.operation.ReplicationChecker;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import javax.sql.DataSource;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static uk.gov.hmcts.reform.wataskmanagementapi.cft.enums.CFTTaskState.UNASSIGNED;
+
+@ActiveProfiles("replica")
+public class ReplicationCheckerTest extends SpringBootIntegrationBaseTest {
+
+    @Autowired
+    TaskResourceRepository taskResourceRepository;
+
+    @Autowired
+    DataSource dataSource;
+
+    @Autowired
+    TaskHistoryResourceRepository taskHistoryResourceRepository;
+    @Autowired
+    ReportableTaskRepository reportableTaskRepository;
+    @Autowired
+    TaskAssignmentsRepository taskAssignmentsRepository;
+
+    SubscriptionCreator subscriptionCreator;
+
+    @Autowired
+    TCExtendedContainerDatabaseDriver tcDriver;
+
+    @Autowired
+    ReplicationChecker replicationChecker;
+
+    @Value("${spring.datasource.jdbcUrl}")
+    private String primaryJdbcUrl;
+
+
+    @Value("${spring.datasource-replica.jdbcUrl}")
+    private String replicaJdbcUrl;
+
+    CFTTaskDatabaseService cftTaskDatabaseService;
+    MIReportingService miReportingService;
+
+    JdbcDatabaseContainer container;
+    JdbcDatabaseContainer containerReplica;
+
+    @BeforeEach
+    void setUp() {
+        subscriptionCreator = new SubscriptionCreator("repl_user", "repl_password",
+            "wa_user", "wa_password");
+        miReportingService = new MIReportingService(taskHistoryResourceRepository, taskResourceRepository,
+            reportableTaskRepository,
+            taskAssignmentsRepository,
+            subscriptionCreator);
+        CFTTaskMapper cftTaskMapper = new CFTTaskMapper(new ObjectMapper());
+        cftTaskDatabaseService = new CFTTaskDatabaseService(taskResourceRepository, cftTaskMapper);
+
+        container = TCExtendedContainerDatabaseDriver.getContainer(primaryJdbcUrl);
+        containerReplica = TCExtendedContainerDatabaseDriver.getContainer(replicaJdbcUrl);
+        Testcontainers.exposeHostPorts(container.getFirstMappedPort(), containerReplica.getFirstMappedPort());
+    }
+
+    @AfterAll
+    void tearDown() {
+        container.stop();
+        containerReplica.stop();
+    }
+
+    @Test
+    void should_save_task_and_get_task_from_replica_tables() {
```

Comment:
Could we add a failure scenario where Tasks are not present in the history?

Reply:
I looked into this, But this is bit trcky as all the records that we insert into task table will be replicated into history table.  Could not have a proper way of stoping this replication. So we cannot have records in the tasks table but not in the history table to simulate the error scenario in IT. But all the failure logics tested in unit tests.

Reply:
Ok that's fine for now as I see you've got good unit test coverage already.

FYI It is possible in an IT to stop replication (e.g. remove the subscription from replica).